catalyst-gateway/tests/Earthfile

```diff
@@ -1,26 +1,89 @@
 VERSION 0.7
 
+build:
+    FROM ../+builder
+    RUN cargo build -p cat-gateway --release
+    SAVE ARTIFACT target/release/cat-gateway cat-gateway
+
+cat-gateway-publish:
+    FROM alpine:3.19
+    ARG tag="latest"
+    ARG address
+    ARG db_url
+    ARG log_level="error"
+    RUN apk upgrade --no-cache && apk add --no-cache gcc
+    COPY +build/cat-gateway .
+    ENTRYPOINT ./cat-gateway run --address $address --database-url $db_url --log-level $log_level
+    SAVE IMAGE cat-gateway:$tag
+
+schemathesis-publish:
+    FROM python:3.12-alpine3.19
+    ARG tag="latest"
+    ARG max_examples=1000
+    ARG max_response_time=200
+    ARG wait_for_schema=15
+    ARG workers=1
+    ARG schema_version=30
+    ARG openapi_spec
+    RUN apk add --no-cache gcc musl-dev
+    RUN python -m pip install schemathesis
+    RUN mkdir /results
+    VOLUME /results
+    ENTRYPOINT st run --checks all $openapi_spec --data-generation-method all --workers=$workers --wait-for-schema=$wait_for_schema \
+                        --max-response-time=$max_response_time --hypothesis-max-examples=$max_examples \
+                                --force-color --skip-deprecated-operations --show-trace \
+                                 --force-schema-version=$schema_version --junit-xml /results/junit-report.xml
+    SAVE IMAGE schemathesis:$tag
+
+fuzzer-api-test:
+    FROM earthly/dind:alpine
+    COPY integration/docker-compose.yml .
+    WITH DOCKER \
+        --compose docker-compose.yml \
+        --load schemathesis:latest=(+schemathesis-publish --openapi_spec="http://127.0.0.1:3030/docs/cat-gateway.json") \
+        --load event-db:latest=(../event-db+build --with_historic_data=false) \
+        --load cat-gateway:latest=(+cat-gateway-publish --address="127.0.0.1:3030" \
+                                        --db_url="postgres://catalyst-event-dev:CHANGE_ME@localhost/CatalystEventDev") \
+        --service event-db \
+        --service cat-gateway \
+        --allow-privileged
+        RUN docker run --net=host --name=st schemathesis:latest || echo fail > fail && \
+                docker logs cat-gateway > ./cat-gateway.log && \
+                    docker cp st:/results/junit-report.xml junit-report.xml
+    END
+    WAIT
+        SAVE ARTIFACT junit-report.xml AS LOCAL junit-report.xml
+        SAVE ARTIFACT cat-gateway.log AS LOCAL cat-gateway.log
+    END
+    IF [ -f fail ]
+        RUN echo "Schemathesis run failed" && \
+            exit 1
+    END
+
 integration-test:
```

Comment:
Should be a `test` target:

```
test-integration:
```

Reply:
This is done to allow us to choose which test target to run by calling it from the `+test` in the main earthfile. Does that make sense to you?